sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -233,13 +245,19 @@ def validate(
         :return: All validation errors
         :type: ValidationResult
         """
-        return self._validate(component, raise_on_failure=raise_on_failure)
+        return self._validate(
+            component,
+            raise_on_failure=raise_on_failure,
+            # TODO 2330505: change this to True after remote validation is ready
+            skip_remote_validation=kwargs.pop("skip_remote_validation", True),
+        )
 
     @monitor_with_telemetry_mixin(logger, "Component.Validate", ActivityType.INTERNALCALL)
     def _validate(  # pylint: disable=no-self-use
         self,
         component: Union[Component, types.FunctionType],
-        raise_on_failure: bool = False,
+        raise_on_failure: bool,
+        skip_remote_validation: bool,
```

Comment:
Will we have same argument on CLI experience?

Reply:
Maybe add it as a private argument after server-side is ready. As you can see, `_validate` is a private interface and I haven't made `skip_remote_validation` a public argument for `component.validate` too. Not sure if we should expose this to customers.